unit-tests/src/test/scala/scala/scalanative/posix/sys/WaitTest.scala

```diff
@@ -0,0 +1,58 @@
+package scala.scalanative.posix.sys
+
+import scala.scalanative.posix.sys.wait._
+import scala.scalanative.posix.unistd.vfork
+import scala.scalanative.unsafe._
+
+import org.junit.Test
+import org.junit.Assert._
+
+class WaitTest {
+  @Test def waitWaitsChildExit(): Unit = {
+    val statLoc = stackalloc[CInt](1)
+
+    def blackHole(a: Any): Unit = ()
+
+    /** these are just here to test the bindings compile and link **/
+    blackHole(WCONTINUED)
+    blackHole(WNOHANG)
+    blackHole(WUNTRACED)
+
+    WIFSIGNALED(!statLoc)
+    WTERMSIG(!statLoc)
+    WIFSTOPPED(!statLoc)
+    WSTOPSIG(!statLoc)
+    WIFCONTINUED(!statLoc)
+
+    //done
+
+    val pid = vfork()
+    if (pid == 0) {
+      sys.exit(0)
+    } else if (pid > 0) {
+      val childPid = waitpid(statLoc)
+
+      assert(
+        pid == childPid,
+        "child that was observed exiting was not the child spawned by fork")
+      assert(WIFEXITED(!statLoc) == 1, "child exited abnormally")
+    } else {
+      assert(false, "forking failed")
```

Comment:
Please consider reporting errno in the "forking failed".  
The message at present is better than none at all, but
is somewhat of a raspberry (insulting noise) in that
it says that it knows why it , but is not going to give you 
a clue.  Sometimes that clue is enough to tell someone
with a failing test what went wrong.  And sometimes errno
is equally inscrutable (i.e. I know but will not tell you...).

That means zeroing errno before the `vfork()` call.

I believe there are examples in the *Test.scala code base.
errno handling is somewhat of a pain (I think must use errno.errno
for arcane reasons) and I think there are two errnos.  
The code examples should give the proper include and
show the usage.  Please ignore the fingerprints of the guilty in
those/that files/file.

Let me know if I am being too terse.  Sorry,m I have a couple of other fires 
burning.

Reply:
done

Reply:
done